gcm/gcm.py

```diff
@@ -284,7 +285,10 @@ def make_request(self, data, is_json=True):
         GCM.log('Request data: {0}', data)
         GCM.log('Request is_json: {0}', is_json)
 
-        response = requests.post(
+        if not session:
+            session = requests.Session()
```

Comment:
Yes you're right it should call close() before exit. I will fix it. (in practice it's unlikely that you'll actually hit a resource limit as the Session and pool are cleaned up by GC)